magento/pwa-studio
# PWA-1683 - My Account: Multi-Wishlist: clean-up

## Description
My Account: Multi-Wishlist: clean-up

## Related Issue
[PWA-1683] My Account: Multi-Wishlist: clean-up


## Acceptance
- Login toasts are rendered if user clicks to add to wishlist but is not signed in
- If EE only one wishlist is allowed, no dialog (to choose a wishlist) should appear
- The "general" Wishlist Button component takes props such as localized "buttonText" string for customization
- The "general" Wishlist Button component takes props for beforeAdd and afterAdd callbacks

### Verification Stakeholders

### Related Issue

### Specification

- Wishlist is enabled in Magento Admin Panel > Stores > Settings > Configuration > Customers > Wish List
- Multi-wishlist is not enabled in Stores > Settings > Configuration > Customers > Wish List > Enable Multiple Wish Lists (EE) OR Maximum number of wishlists is set to one (1)

## Test Plan






#### Test scenario(s) for direct fix/feature

- Create a new wishlist
- Verify that you are not prompted to set its "visibility" (pubic / private)
- Edit an existing wishlist
- Verify that you are not prompted to set its "visibility" (public / private)
- Verify that you can create as many wishlists as are set in the backend (Stores > Settings > Configuration > Customers > Wish List)
- Verify that attempting to add another wishlist after reaching that threshold presents an error message and lets you recover gracefully
- Add many products to the wishlist (20+) and verify
- Verify that the first wishlist in the list is the most recently created
- Verify that the first wishlist is expanded by default. If there is a second list it is collapsed by default.
- Verify the lists have text in the header of the List container(s) that displays the number of visible items and the total number of items in the list.
- Verify that if there are more than 20 items saved to a List, a "Load More" button appears at the bottom after the first 20 (visible) items.
- Verify that when user taps the "Load More" button the next 'batch' of saved items (1-20 items) should appear in row(s) beneath the first 20.
- Verify that text in the container header updates to include next 20 loaded items in the count - ( ex. "Showing 20 of 58 items in this list" -> "Showing 40 of 58 items in this list" )
- Verify that when there are no more items in the list the "Load More" button doesn't appear;
- Verify that items are returned in the order of most recently added items
- Verify that clicking the "load more" button loads more wishlist items
- Verify that the "load more" button remains visible (after the items) until all items have been loaded
- Verify that for each product you can add to cart and remove from wishlist

#### Test scenario(s) for any existing impacted features/areas
Example
-   Verify user is able to sort data after applying filters on category page
-   Verify user is able to sort data after applying filters on search page

#### Test scenario(s) for any Magento Backend Supported Configurations
Example
- Update default Sort value in backend and repeat above scenarios

#### Is Browser/Device testing needed?
Example
- Yes, browser testing is needed as X UI component may be impacted on <browser>
- Yes, device testing is needed as X UI component may be impacted on <mobile|desktop|etc>

#### Any ad-hoc/edge case scenarios that need to be considered?
Example
- Apply all filters to get 0 results and then remove filters to see respective products 

## Screenshots / Screen Captures (if appropriate)

## Breaking Changes (if any)





## Checklist



-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.
